bgpd/bgp_nb_config.c

```diff
@@ -120,29 +315,32 @@ int routing_control_plane_protocols_control_plane_protocol_bgp_global_router_id_
 int routing_control_plane_protocols_control_plane_protocol_bgp_global_confederation_identifier_modify(
 	struct nb_cb_modify_args *args)
 {
-	switch (args->event) {
-	case NB_EV_VALIDATE:
-	case NB_EV_PREPARE:
-	case NB_EV_ABORT:
-	case NB_EV_APPLY:
-		/* TODO: implement me. */
-		break;
-	}
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	struct bgp *bgp;
+	as_t as;
+
+	bgp = nb_running_get_entry(args->dnode, NULL, true);
+
+	as = yang_dnode_get_uint32(args->dnode, NULL);
+
+	bgp_confederation_id_set(bgp, as);
```

Comment:
Added (as == 0) check into validation phase.

Reply:
Now that you did that it would be nice to remove that check from `bgp_confederation_id_set()` since it became redundant. You can change the return type of that function to `void` as well.

Reply:
done.